### PR TITLE
Second task from Netwroking

### PR DESCRIPTION
**Pull request by p-nedelchev**
Needs review

### CODE REVIEW

File: src/main/java/com/clouway/second/client/ClientDemo.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.clouway.second.server.Server;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+/**
+ * @author Petar Nedelchev <peter.krasimirov@gmail.com>
+ */
+public class ClientDemo {
```
Comment (tsony-tsonev) [inline src/main/java/com/clouway/second/client/ClientDemo.java]: you dont need demo when you have tests, you can delete it

File: src/main/java/com/clouway/second/server/ServerDemo.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.clouway.second.CalendarUtil;
+
+import java.net.InetAddress;
+import java.util.Calendar;
+
+/**
+ * @author Petar Nedelchev <peter.krasimirov@gmail.com>
+ */
+public class ServerDemo {
```
Comment (tsony-tsonev) [inline src/main/java/com/clouway/second/server/ServerDemo.java]: same here

File: src/test/java/com/clouway/second/ClientTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    private String messageSentByServer;
+
+
+
+    @Before
+    public void setUp() throws Exception {
+        address = InetAddress.getByName("127.0.0.1");
+        client = new Client(address, 15000);
+        messageSentByServer = "Hello! ";
+        fakeServer = new FakeServer(15000, messageSentByServer, new CalendarUtil() {
```
Comment (tsony-tsonev) [inline src/test/java/com/clouway/second/ClientTest.java]: you can extract the fake calendar for better readability in the tests

File: src/test/java/com/clouway/second/ServerTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    private InetAddress localhost;
+    private FakeClient fakeClient;
+    private Date time;
+
+
+
+    @Before
+    public void setUp() throws Exception {
+        localhost = InetAddress.getByName("127.0.0.1");
+        server = new Server(15000, new CalendarUtil() {
```
Comment (tsony-tsonev) [inline src/test/java/com/clouway/second/ServerTest.java]: same here

File: src/test/java/com/clouway/second/ClientTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                return time.toString();
+            }
+        });
+    }
+
+    @Test
+    public void happyPath() throws Exception {
+        new Thread(client).start();
+        fakeServer.sendMessage();
+        String expectedMessage = client.getLastReceivedMessage();
```
Comment (tsony-tsonev) [inline src/test/java/com/clouway/second/ClientTest.java]: this is the actual message. also You can pass the expected message to the sendMessage method to improve the test readability

**Comment by tsony-tsonev** [comment]
LGTM after you fix the comments

File: src/main/java/com/clouway/second/client/ClientDemo.java (hunk 1 above)
Comment (p-nedelchev) [inline src/main/java/com/clouway/second/client/ClientDemo.java]: Demo Removed

File: src/main/java/com/clouway/second/server/ServerDemo.java (hunk 2 above)
Comment (p-nedelchev) [inline src/main/java/com/clouway/second/server/ServerDemo.java]: Demo Removed

File: src/test/java/com/clouway/second/ClientTest.java (hunk 3 above)
Comment (p-nedelchev) [inline src/test/java/com/clouway/second/ClientTest.java]: Done

File: src/test/java/com/clouway/second/ServerTest.java (hunk 4 above)
Comment (p-nedelchev) [inline src/test/java/com/clouway/second/ServerTest.java]: Done

**Pull request merged**